include/bluetooth/mesh/light_ctrl_srv.h

```diff
@@ -112,18 +129,22 @@ struct bt_mesh_light_ctrl_srv_reg_cfg {
 	uint8_t accuracy;
 };
 
-/** Illumination regulator */
+struct bt_mesh_light_ctrl_srv_reg_cb;
+
+/** Light Lightness Control regulator interface */
 struct bt_mesh_light_ctrl_srv_reg {
-	/** Regulator step timer */
-	struct k_work_delayable timer;
-	/** Internal integral sum. */
-	float i;
-	/** Previous output */
-	uint16_t prev;
-	/** Regulator configuration */
+	void *user_data;
+	struct bt_mesh_light_ctrl_srv_reg_cb *cb;
 	struct bt_mesh_light_ctrl_srv_reg_cfg cfg;
 };
 
+/** Light Lightness Control regulator callbacks */
+struct bt_mesh_light_ctrl_srv_reg_cb {
+	void (*init)(struct bt_mesh_light_ctrl_srv *srv, void *user_data);
+	void (*start)(struct bt_mesh_light_ctrl_srv *srv, void *user_data);
+	void (*stop)(struct bt_mesh_light_ctrl_srv *srv, void *user_data);
+};
```

Comment:
Is it possible to move the entire definition of `struct bt_mesh_light_ctrl_srv_reg_cb` to the top? I can't see any particular reason for declaring it first.

Reply:
You are correct, this is a holdover from when it took a `..._srv_reg *` as an argument :) Will move.